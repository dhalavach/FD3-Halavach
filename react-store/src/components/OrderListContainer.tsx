import { Product, orderObject } from "../types/Types";

export default function OrderListContainer(props: { order: orderObject }) {
  const { order } = props;
  return (
    <div className="orders-list__order-container" key={order.id}>
      <ul>
        <li>Name: {order.name}</li>
        <li>Email: {order.email}</li>
        <li>Address: {order.address}</li>
        <li>
          {order.orderedProducts.map((p: Product) => {
            return (
              <div key={order.id.toString() + p.itemName}>
                {p.itemName} x {p.count}
              </div>
            );
          })}
        </li>
      </ul>
    </div>
  );
}
