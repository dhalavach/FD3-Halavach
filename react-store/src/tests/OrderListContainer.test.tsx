import '@testing-library/jest-dom';
import OrderListContainer from '../components/OrderListContainer';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../components/Store';
import { BrowserRouter as Router } from 'react-router-dom';

const mockOrder = {
  id: 1,
  name: 'John Doe',
  email: 'foo@gmail.com',
  address: 'Some Street 7, 1414',
  orderedProducts: [
    {
      id: 2,
      className: 'Item',
      itemType: 'motherboard',
      itemName: 'ASUS PRIME Z790-P D4',
      itemPrice: 820,
      itemImage: '/assets/images/asusz790.jpg',
      itemImageURL: 'https://uni.by/files/ItemsImages/135354_r300.jpg',
      itemImageAlt: 'Asus Z790 Prime',
      itemQuantity: 1,
      itemDescription: 'Lorem ipsum...',
      count: 1,
    },
    {
      id: 14,
      className: 'Item',
      itemType: 'CPU',
      itemName: 'Intel 13700k',
      itemPrice: 699,
      itemImage: '/assets/images/13700k.jpg',
      itemImageURL: 'asdf',
      itemImageAlt: '13700k',
      itemQuantity: 7,
      itemDescription:
        's-1700| 16C/24T (8P 3.4/5.4GHz + 8E 2.5/4.2GHz) 30MB 253W Intel UHD 770',
      count: 1,
    },
  ],
};

test('component renders without throwing an error', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <OrderListContainer order={mockOrder} />{' '}
        </Router>
      </Provider>
    )
    .toJSON();
  expect(tree).toBeTruthy();
});

test('component matches the snapshot', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <OrderListContainer order={mockOrder} />{' '}
        </Router>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
