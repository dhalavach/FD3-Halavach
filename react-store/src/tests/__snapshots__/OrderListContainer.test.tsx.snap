// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component matches the snapshot 1`] = `
[
  <div
    className="orders-list__order-container"
  >
    <ul>
      <li>
        Name: 
        John Doe
      </li>
      <li>
        Email: 
        foo@gmail.com
      </li>
      <li>
        Address: 
        Some Street 7, 1414
      </li>
      <li>
        <div>
          ASUS PRIME Z790-P D4
           x 
          1
        </div>
        <div>
          Intel 13700k
           x 
          1
        </div>
      </li>
    </ul>
  </div>,
  " ",
]
`;
